### ISSUE TITLE
Rule Request: Protocol conformance order

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#2751]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

> 1. Why should this rule be added? Share links to existing discussion about what the community thinks about this.

Similar to the [`modified_order` rule](https://github.com/realm/SwiftLint/blob/master/Rules.md#modifier-order), it would be useful to be able to enforce the order of built-in protocols in a conformance. And also enforce that built-in protocols come first.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
struct Foo: RawRepresentable, Hashable {}
```

```swift
struct Foo: CustomProtocol, Hashable {}
```

Would not trigger:

```swift
struct Foo: Hashable, RawRepresentable {}
```

```swift
struct Foo: Hashable, CustomProtocol {}
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

No

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Enabled by default
